subprojects/integ-test/integ-test.gradle

```diff
@@ -35,6 +35,7 @@ testFixtures {
 
 integTestTasks.configureEach { task ->
     task.libsRepository.required = true
+    task.gradleInstallationForTest.samplesRequired.set(true)
```

Comment:
I think the coexistence of this method and [this method](https://github.com/gradle/gradle/blob/938ded09c57a4f76503c7fe9c8e6d34082bd1930/subprojects/distributions/distributions.gradle#L152) makes thing complicated. IMHO, we should not expose `binaryDistributions` of `DistributionTest` to the outside world. Instead, we expose public method `requireAllDistribution()` and `requireBinDistribution()` and don't allow direct manipulation of `BinaryDistributions` internal state. 

This would also get rid of current complicated provider stuff `project.objects.property(Boolean::class.javaObjectType)` and `project.provider { binaryDistributions.distributionsRequired }`. WDYT?

Reply:
You are probably right, we should do something similar as we did for the `PerformanceTest` task: Expose everything which needs to be configured on the task itself and then have a non-static inner class as command line argument provider which may declare additional/normalized inputs and sets the system properties.